piccolo_api/mfa/email/tables.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+
+import datetime
+
+from piccolo.columns import Email, Timestamptz, Varchar
+from piccolo.table import Table
+
+
+class EmailCode(Table):
+    email = Email()
+    code = Varchar()  # TODO - look how best to generate the codes
+    created_at = Timestamptz()
+    used_at = Timestamptz(null=True, default=None)
+
+    _expiry_time = datetime.timedelta(minutes=5)
+
+    @classmethod
+    async def create_new(cls, email: str) -> EmailCode:
+        # TODO - generate proper code
```

Comment:
Good idea.

I had a thought about the email implementation. At the moment it has its own database table, but we could potentially use the `AuthenticatorSecret` table.

My original idea with `EmailCode` is a new one would be generated each time the user logs in.

If instead we used the `AuthenticatorSecret` table, we could generate a secret (like with authenticator apps), but rather than sharing that secret with the client, each time they login we just generate a TOTP code using the secret and send it to them via email. When the client submits the code, we verify it using the secret.